### PR TITLE
fix: handle wildcard parent patterns

### PR DESCRIPTION
**Pull request by miraleung**
The monolith generator currently NPEs when it encounters `pattern: "*"` in conjunction with actual patterns, such as with [AlertPolicy](https://github.com/googleapis/googleapis/blob/master/google/monitoring/v3/alert.proto#L44).

### CODE REVIEW

File: src/main/java/com/google/api/codegen/config/ResourceDescriptorConfig.java [hunk 1]
```diff
@@ -75,6 +76,19 @@
   /** The entity name for the resource config. */
   public abstract String getDerivedEntityName();
 
+  public static ResourceDescriptorConfig getWildcardResource(ProtoFile protoFile) {
+    return new AutoValue_ResourceDescriptorConfig(
+        /* isDefinedAtMessageLevel= */ false,
```
Comment (vam-google) [inline src/main/java/com/google/api/codegen/config/ResourceDescriptorConfig.java:81]: super not important: please format these (some have spaces after `/*` some don't)

File: src/test/java/com/google/api/codegen/config/ResourceDescriptorConfigTest.java [hunk 2]
```diff
@@ -294,6 +294,61 @@ public void testGetChildParentResourceMap_MultiParentsMultiPatterns() {
...
+
+    Map<String, ResourceDescriptorConfig> descriptorConfigMap =
+        getDescriptorConfigMap(sauce, meat, veggie);
+    Map<String, List<ResourceDescriptorConfig>> patternResourceDescriptorMap =
+        ResourceDescriptorConfig.getPatternResourceMap(Arrays.asList(sauce, meat, veggie));
+
+    Map<String, List<ResourceDescriptorConfig>> childParentResourceMap =
+        ResourceDescriptorConfig.getChildParentResourceMap(
+            descriptorConfigMap, patternResourceDescriptorMap);
+    System.out.println(
```
Comment (vam-google) [inline src/test/java/com/google/api/codegen/config/ResourceDescriptorConfigTest.java:344]: Is it a debug artifact?

File: src/main/java/com/google/api/codegen/config/ResourceDescriptorConfig.java [hunk 3]
```diff
@@ -218,7 +232,25 @@ private String getUnqualifiedTypeName() {
         ImmutableList.copyOf(descriptorConfigMap.values());
     for (Map.Entry<String, ResourceDescriptorConfig> entry : descriptorConfigMap.entrySet()) {
       ResourceDescriptorConfig childResource = entry.getValue();
+      System.out.println(
+          "DEL: Looking at entry "
```
Comment (vam-google) [inline src/main/java/com/google/api/codegen/config/ResourceDescriptorConfig.java:236]: Are these intended, or are they the debug artifacts?

Comment (miraleung) [reply src/main/java/com/google/api/codegen/config/ResourceDescriptorConfig.java:236]: Removed.

File: src/test/java/com/google/api/codegen/config/ResourceDescriptorConfigTest.java (hunk 2 above)
Comment (miraleung) [reply src/test/java/com/google/api/codegen/config/ResourceDescriptorConfigTest.java:344]: Removed.

File: src/main/java/com/google/api/codegen/config/ResourceDescriptorConfig.java (hunk 1 above)
Comment (miraleung) [reply src/main/java/com/google/api/codegen/config/ResourceDescriptorConfig.java:81]: Done.

**Pull request merged**